src/solver.jl

```diff
@@ -1,159 +1,67 @@
+        # @assert !isempty(vertices)
 
         mis_count = Vector{CountingMIS}(undef, length(dnf.clauses))
         
         @threads for i in 1:length(dnf.clauses)
             clause = dnf.clauses[i]
-            removed_vertices = rv(vertices, g, clause)
+            rvs = removed_vertices(vertices, g, clause)
             gi = copy(g)
-            rem_vertices!(gi, removed_vertices)
-            @assert !isempty(removed_vertices)
-            mis_count[i] = mis_solver(gi, strategy, kneighbor, use_rv, use_filter) + count_ones(clause.val)
+            rem_vertices!(gi, rvs)
+            # @assert !isempty(rvs)
+            mis_count[i] = mis_solver(gi, strategy, measurement, vertex_select, filter) + count_ones(clause.val)
         end
-        max_mis = max(mis_count...)
+        max_mis = maximum(mis_count)
 
         return max_mis
     end
 end
 
-function neighbor_cover(g::SimpleGraph, v::Int, k::Int)
-    @assert k >= 0
-    vertices = [v]
-    for _ = 1:k
-        vertices = union(vertices, (neighbors(g, w) for w in vertices)...)
-    end
-    openvertices = [v for v in vertices if !all(x->x ∈ vertices, neighbors(g, v))]
-    return vertices, openvertices
-end
-
-"""
-    optimal_branching_dnf(g::SimpleGraph, strategy::BranchingStrategy, kneighbor::Int, use_rv::Bool)
-
-Compute the optimal branching in a directed acyclic graph (DAG) using a specified branching strategy.
-
-# Arguments
-- `g::SimpleGraph`: The input graph.
-- `strategy::BranchingStrategy`: The branching strategy to use.
-- `kneighbor::Int`: The number of neighbors to consider when selecting the branching vertex.
-- `use_rv::Bool`: Whether to use the number of removed vertices during the branching process.
-- `use_filter::Bool`: Whether to use the environment filter.
-
-# Returns
-- `vertices`: The set of vertices selected for the optimal branching.
-- `openvertices`: The set of open vertices in the optimal branching.
-- `impl_strategy`: The implementation strategy used for the optimal branching.
-
-"""
-function optimal_branching_dnf(g::SimpleGraph, strategy::BranchingStrategy, kneighbor::Int, use_rv::Bool, use_filter::Bool)
-    # reference: Exaxt Exponential Algorithms by Fomin and Kratsch, chapter 2.3
-    degree_g = degree(g)
-
-    # use minimum boundary instead
-    vs_min = Int[]
-    ovs_min = [1:nv(g)...]
-    for v in 1:nv(g)
-        vs, ovs = neighbor_cover(g, v, kneighbor)
-        if length(ovs) < length(ovs_min)
-            vs_min = vs
-            ovs_min = ovs
-        end
-    end
-    vertices = vs_min
-    openvertices = ovs_min
+function optimal_branching_dnf(g::SimpleGraph, strategy::AbsractBranching, measurement::AbstractMeasurement, vertex_select::AbstractVertexSelector, filter::AbstractTruthFilter)
+    
+    vertices, openvertices = select_vertex(g, vertex_select)
 
     subg, vmap = induced_subgraph(g, vertices)
     tbl = reduced_alpha_configs(subg, Int[findfirst(==(v), vertices) for v in openvertices])
```

Comment:
Please add an MIS solver option to this function, so that we can switch the backend in the future.

Reply:
Currently this function is used as a backend of the mis solver.